File: python/reverse_string.py
class Solution:
    def reverseString(self, s: list[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        l: int = 0
        r: int = len(s) - 1
        while l < r:
            tmp: str = s[l]
            s[l] = s[r]
            s[r] = tmp
            l += 1
            r -= 1


def test_reverseString_case_1():
    # arrange
    s: list[str] = ["h", "e", "l", "l", "o"]
    expected: list[str] = ["o", "l", "l", "e", "h"]

    # act
    solution = Solution()
    solution.reverseString(s)

    # assert
    assert expected == s

File: rust/src/code/reverse_string.rs
struct Solution {}

impl Solution {
    pub fn reverse_string(s: &mut Vec<char>) {
        let mut l: usize = 0;
        let mut r: usize = s.len() - 1;

        while l < r {
            let tmp: char = s[l];
            s[l] = s[r];
            s[r] = tmp;
            l += 1;
            r -= 1;
        }
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_reverse_string_case_1() {
        // arrange
        let mut s: Vec<char> = vec!['h', 'e', 'l', 'l', 'o'];
        let expected: Vec<char> = vec!['o', 'l', 'l', 'e', 'h'];
        // act
        Solution::reverse_string(&mut s);

        // assert
        assert_eq!(expected, s);
    }
}
